backend/service/k8s/job_test.go

```diff
@@ -54,6 +54,24 @@ func testJobClientset() k8s.Interface {
 	return fake.NewSimpleClientset(job)
 }
 
+func TestDescribeJob(t *testing.T) {
+	cs := testJobClientset()
+
+	s := &svc{
+		manager: &managerImpl{
+			clientsets: map[string]*ctxClientsetImpl{"testing-clientset": {
+				Interface: cs,
+				namespace: "testing-namespace",
+				cluster:   "testing-cluster",
+			}},
+		},
+	}
+
+	cron, err := s.DescribeJob(context.Background(), "testing-clientset", "testing-cluster", "testing-namespace", "testing-job-name")
```

Comment:
can we also add assertions for the not found and found more than 1 job case?

Reply:
Added assertion for not found case and changed implementation to avoid more than 1 job case. Since name is key for the object, duplicate jobs cannot be created.

Reply:
ah yes that's correct! however, the pattern we've historically set is to have our `Describe` APIs internally make a `List` K8s API call. This was due to an issue with the omni search (full details [here](https://github.com/lyft/clutch/issues/257)) failing to look up a resource and the [fix](https://github.com/lyft/clutch/pull/271) was to use fieldSelectors to search cluster wide for a resource's metadata.name.

@mikecutalo - thoughts on this API making a `List` vs `Get` call?

Reply:
Thanks for the background on why `List` is used in place of `Get` in all these APIs. I would revert to the `List` based implementation if that is desired, however, IMHO, using `DescribePod` when the intention is to search across namespaces might not be appropriate. Perhaps, we could call a version of `List` API that allows specifying `FieldSelector` as option.

Reply:
Updated the API implementation to use `List` and changed tests accordingly.

Reply:
thank you for the feedback! will share it with the team.

Reply:
Thank you @scarlettperry! And thank you @sbansal7 for changing over to the list API!